### PR TITLE
fix(externallink): Map font colours to correct media query rules, add better icon colour handling

### PR DESCRIPTION
**Pull request by Pallandor**
- Map font colours to correct media queries
- Implement renderIcon prop/MediaQueryDetector, dynamically hand desired icon fill colour via params, update tests

---
This change is [Reviewable](https://reviewable.io/reviews/qantasairways/runway/28)

### CODE REVIEW

File: src/icons/PinIcon/index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { generate } from 'shortid';
 
-function PinIcon({ height, width, className }) {
+function PinIcon({ height, width, className, color }) {
```
Comment (rachmulvey) [inline src/icons/PinIcon/index.js]: Let's remove the `color` prop and `fill`, the idea is that the fill colour can be overridden using the existing `className` prop

File: styleguide.config.js [hunk 2]
```diff
@@ -43,5 +42,6 @@ module.exports = {
     }
   },
   version,
-  webpackConfig: createConfig([babel(), postcss()])
+  webpackConfig: createConfig([babel(), postcss()]),
+  skipComponentsWithoutExample: true
```
Comment (rachmulvey) [inline styleguide.config.js]: let's remove this as styleguidist config was refactored not to need it

File: src/icons/PinIcon/index.js (hunk 1 above)
Comment (Pallandor) [reply src/icons/PinIcon/index.js]: Fixed. `ExternalLink` now hands down a dynamic className via `renderIcon` on resizing events

File: styleguide.config.js (hunk 2 above)
Comment (Pallandor) [reply styleguide.config.js]: Fixed (removed).

**Pull request merged**